model-optimizer/extensions/middle/ssd_anchors_middle_replacer.py

```diff
@@ -0,0 +1,203 @@
+        slice_like.out_port(0).disconnect()
+        match['reshape3'].out_port(0).get_connection().set_source(slice_like.out_port(0))
+        slice_like['override_output_shape'] = True
+
+        reshape = create_op_node_with_second_input(graph, Reshape, int64_array([1, -1]), op_attrs=dict(
+            name=slice_like_name + '/Reshape', override_output_shape=True))
+        slice_like.out_port(0).get_connection().insert_node(reshape)
+
+
+class SsdAnchorsMiddleReplacer(MiddleReplacementPattern):
+
+    """
+    Replacing sub-graph with all anchors to sub-graph which calculates prior boxes values by formulas:
+
+    value[i] = xmin = value[i] - (value[i + 2] / 2)
+    value[i + 1] = ymin = value[i + 1] - (value[i + 3] / 2)
+    value[i + 2] = xmax = value[i] + (value[i + 2] / 2)
+    value[i + 3] = ymax = value[i + 1] + (value[i + 3] / 2)
+    """
+
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['fw'] == 'mxnet' and graph.graph['cmd_params'].enable_ssd_gluoncv]
+
+    def run_after(self):
+        return [SliceLikeToStridedSlice, SsdAnchorMiddleReshape]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('slice_like', dict(op='StridedSlice')),
+                ('slice_like_data', dict(kind='data')),
+                ('reshape0', dict(op='Reshape')),
+                ('reshape0_data', dict(kind='data')),
+                ('concat', dict(op='Concat')),
+            ],
+            edges=[
+                ('slice_like', 'slice_like_data'),
+                ('slice_like_data', 'reshape0'),
+                ('reshape0', 'reshape0_data'),
+                ('reshape0_data', 'concat', {'in': 0}),
+        ])
+
+    def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
+        concat_node = match['concat']
+        concat_name = concat_node.soft_get('name', concat_node.id)
+        if len(concat_node.out_nodes()[0].out_nodes()) == 0:
+            return
+
+        detection_output_port = concat_node.out_port(0).get_destination()
+        concat_node.out_port(0).disconnect()
+
+        concat_reshape = create_op_node_with_second_input(graph, Reshape, int64_array([1, 2, -1]), op_attrs=dict(
+            name=concat_name + '/Reshape'), input_node=concat_node)
+        split_node = create_op_node_with_second_input(graph, Split, int64_array(1), op_attrs=dict(
+            name=concat_name + '/Split', num_splits=2), input_node=concat_reshape)
+        split_node_reshape = create_op_node_with_second_input(graph, Reshape, int64_array([-1, 4]), op_attrs=dict(
+            name=split_node.name + '/Reshape'))
+        split_node.out_port(0).connect(split_node_reshape.in_port(0))
+        value = create_op_node_with_second_input(graph, Split, int64_array(1), op_attrs=dict(
+            name=split_node_reshape.name + '/Split', num_splits=4), input_node=split_node_reshape)
+
+        #xmin
+        xmin = Sub(graph, dict(name=value.name + '/Sub_1')).create_node()
```

Comment:
You can move lines 166-171 to a separate function and re-use it in lines 174-179